src/main/java/edu/iis/powp/decorator/FlipPlotterDecorator.java

```diff
@@ -23,7 +23,10 @@ public FlipHorizontalPlotterDecorator(IPlotter originalPlotter) {
      */
     @Override
     public void drawTo(int i, int i1) {
-        i *= scale;
+        if(super.transformation.equalsIgnoreCase("flipHorizontal"))
```

Comment:
This solution is entirely unappealing. Amount of duplicated code has been augmented with if instructions (always the same code in drawTo and setPosition).
originalPlotter.drawTo(i, i1); and   originalPlotter.setPosition(i, i1); still duplicated in all decorators.